source/ShipJumpNavigation.cpp

```diff
@@ -0,0 +1,223 @@
+/* ShipJumpNavigation.cpp
+Copyright (c) 2022 by Amazinite
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "ShipJumpNavigation.h"
+
+#include "Outfit.h"
+#include "Ship.h"
+#include "System.h"
+
+#include <algorithm>
+#include <cassert>
+#include <iterator>
+
+using namespace std;
+
+const double ShipJumpNavigation::DEFAULT_HYPERDRIVE_COST = 100.;
+const double ShipJumpNavigation::DEFAULT_SCRAM_DRIVE_COST = 150.;
+const double ShipJumpNavigation::DEFAULT_JUMP_DRIVE_COST = 200.;
+
+
+
+// Set the owner of this jump navigation, storing a pointer to it for later use.
+void ShipJumpNavigation::SetOwner(const Ship *ship)
+{
+	this->ship = ship;
+}
+
+
+
+// Calibrate this ship's jump navigation information, caching its jump costs, range, and capabilities.
+void ShipJumpNavigation::Calibrate()
+{
+	assert(ship && "A jump navigation's ship cannot be null.");
+	Outfit attributes = ship->Attributes();
```

Comment:
Shouldn't this be `const` and a reference?
```suggestion
	const Outfit &attributes = ship->Attributes();
```